Python/thread_nt.h

```diff
@@ -207,16 +206,35 @@ PyThread_start_new_thread(void (*func)(void *), void *arg)
         /* I've seen errno == EAGAIN here, which means "there are
          * too many threads".
          */
-        int e = errno;
-        threadID = (unsigned)-1;
         HeapFree(GetProcessHeap(), 0, obj);
+        return PYTHREAD_INVALID_THREAD_ID;
     }
-    else {
-        CloseHandle(hThread);
-    }
+    *handle = (Py_uintptr_t) hThread;
+    return threadID;
+}
+
+unsigned long
+PyThread_start_new_thread(void (*func)(void *), void *arg) {
+    Py_uintptr_t handle;
+    unsigned long threadID = PyThread_start_joinable_thread(func, arg, &handle);
+    CloseHandle((HANDLE) handle);
     return threadID;
 }
 
+int
+PyThread_join_thread(Py_uintptr_t handle) {
+    HANDLE hThread = (HANDLE) handle;
+    int errored = (WaitForSingleObject(hThread, INFINITE) != WAIT_OBJECT_0);
```

Comment:
We probably want to include the sigint event in here as well, so that Ctrl+C can abort a join (e.g. https://github.com/python/cpython/blob/main/Modules/timemodule.c#L2259 )

Reply:
There are two reasons not to do it:
1. [pthread_join](https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_join.html) cannot be interrupted (_"The pthread_join() function shall not return an error code of [EINTR]"_)
2. If you look at how this is used in `threading.py`, `PyThread_join_thread` is only called when the thread state lock can be taken, i.e. when the thread is exiting. This is just to make sure that when `Thread.join()` returns, the OS thread has also exited and its resources have been released.

Reply:
Maybe label/document the function as uninterruptible then? Before we try and use it in a different place where it would cause trouble. (Also, one of the thread locks is uninterruptible and that annoys users, I'm not sure exactly which one, but it sounds like it wouldn't be this one.)